R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,23 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (!grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)) {
```

Comment:
Yes, that basically checks if you are working from the sparkR shell or RStudio, and do not download in the former case.